addon/-private/system/model/model.js

```diff
@@ -1308,27 +1308,21 @@ Model.reopenClass({
    @return {Object} the inverse relationship, or null
    */
   inverseFor(name, store) {
-    let relationship = get(this, 'relationshipsByName').get(name);
-    if (!relationship) {
-      return null;
-    }
-
-    let options = relationship.options;
-    if (options && options.inverse === null) {
-      return null;
-    }
-
-    return this._inverseFor(name, store);
-  },
-
-  _inverseFor(name, store) {
     let inverseMap = get(this, 'inverseMap');
     if (inverseMap[name]) {
```

Comment:
should this also check for `null` and if it is `null` return `null`, rather then failing this branch and looking it up again on 1315 only to find `null` again?